test/unit/local/analysis.js

```diff
@@ -466,17 +467,50 @@ describe('core/local/analysis', function () {
     ])
   })
 
+  it('sort correctly unlink + add + move dir', () => {
+    const dirStats = {ino: 1}
+    const fileStats = {ino: 2}
+    const newFileStats = {ino: 3}
+
+    const oldDirPath = 'root/src/dir'
+    const oldFilePath = 'root/src/dir/file.rtf'
+    const newDirPath = 'root/dir/file.rtf'
+    const newFilePath = 'root/dir/file.rtf'
+
+    const dirMetadata /*: Metadata */ = metadataBuilders.dir().path(oldDirPath).ino(dirStats.ino).build()
+    const fileMetadata  /*: Metadata */ = metadataBuilders.file().path(oldFilePath).ino(fileStats.ino).build()
+
+    const events /*: LocalEvent[] */ = [
+      {type: 'addDir', path: newDirPath, stats: dirStats},
+      {type: 'add', path: newFilePath, stats: newFileStats},
+      {type: 'unlinkDir', path: oldDirPath, old: dirMetadata},
+      {type: 'unlink', path: oldFilePath, old: fileMetadata}
+    ]
+    const pendingChanges /*: LocalChange[] */ = []
+
+    const changes = analysis(events, pendingChanges)
+    changes.map(change => change.type).should.deepEqual([
+      'DirMove', 'FileAddition', 'FileDeletion'
+    ])
+  })
+
   it('sorts actions', () => {
+    const normalizer = (x) => {
+      x.path = path.normalize(x.path)
+      if (x.old) x.old.path = path.normalize(x.old.path)
+      return x
+    }
```

Comment:
This would probably deserve its own `BaseMetadataBuilder` method.
Can be done in another PR.
Although since `old` is not necessarily cloned when passed to the builder, mutating it on the whole test codebase may introduce subtle bugs...